python/ray/data/_internal/logical/operators/from_numpy_operator.py

```diff
@@ -0,0 +1,19 @@
+from typing import TYPE_CHECKING, List, Union
+
+from ray.data._internal.logical.interfaces import LogicalOperator
+from ray.types import ObjectRef
+
+if TYPE_CHECKING:
+    import numpy as np
+
+
+class FromNumpyRefs(LogicalOperator):
+    """Logical operator for `from_numpy_refs`."""
+
+    def __init__(
+        self,
+        ndarrays: Union[List[ObjectRef["np.ndarray"]], List["np.ndarray"]],
```

Comment:
Or perhaps from tf / torch should extend this class instead?

Reply:
yeah I think `from_tf` can extend this class.

`from_torch` seems not guarantee to return NumPy ndarray, so it looks to me okay to extend `FromItems`.